crm_deduplicate_acl/wizards/partner_merge.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# © 2016 Tecnativa, S.L. - Jairo Llopis
+# © 2016 Tecnativa, S.L. - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, SUPERUSER_ID, models
+from odoo.exceptions import ValidationError, UserError
+import logging
+
+
+_logger = logging.getLogger('base.partner.merge')
+
+
+class BasePartnerMergeAutomaticWizard(models.TransientModel):
+    _inherit = "base.partner.merge.automatic.wizard"
+
+    def _merge(self, partner_ids, dst_partner=None):
+        """ private implementation of merge partner
+            :param partner_ids : ids of partner to merge
+            :param dst_partner : record of destination res.partner
+        """
+        Partner = self.env['res.partner']
+        partner_ids = Partner.browse(partner_ids).exists()
+        if len(partner_ids) < 2:
+            return
+
+        if len(partner_ids) > 3:
+            raise UserError(_("For safety reasons, you cannot merge more than"
+                              " 3 contacts together. You can re-open"
+                              " the wizard several times if needed."))
+
+        # check if the list of partners to merge contains child/parent relation
+        child_ids = self.env['res.partner']
+        for partner_id in partner_ids:
+            child_ids |= Partner.search([('id', 'child_of', [partner_id.id])])\
+                         - partner_id
+        if partner_ids & child_ids:
+            raise UserError(_("You cannot merge a contact with one"
+                              " of his parent."))
+
+        user = self.env.user
+
+        if SUPERUSER_ID != self.env.uid and not user.has_group(
+                'crm_deduplicate_acl.group_unrestricted') and\
+                        len(set(partner.email for partner in partner_ids)) > 1:
+            raise UserError(_("All contacts must have the same email."
+                              " Only the Administrator can merge contacts"
+                              " with different emails."))
+
+
+        # remove dst_partner from partners to merge
+        if dst_partner and dst_partner in partner_ids:
+            src_partners = partner_ids - dst_partner
+        else:
+            ordered_partners = self._get_ordered_partner(partner_ids.ids)
+            dst_partner = ordered_partners[-1]
+            src_partners = ordered_partners[:-1]
+        _logger.info("dst_partner: %s", dst_partner.id)
+
+        # FIXME: is it still required to make and exception for
```

Comment:
It is still commented in V10:
https://github.com/odoo/odoo/blob/4b102523b584a5383009f2e19b6f7d0c07c61591/addons/crm/wizard/base_partner_merge.py#L321